src/bpf/bpfd-manager.c

```diff
@@ -0,0 +1,465 @@
+        Manager *m = ASSERT_PTR(userdata);
+        usec_t usec_now;
+        int r;
+
+        assert(s);
+
+        /* Collect BPF tax information */
+        r = log_bpf_maps_and_progs(m);
+        if (r < 0)
+                return r;
+
+        r = sd_event_now(sd_event_source_get_event(s), CLOCK_MONOTONIC, &usec_now);
+        if (r < 0)
+                return log_error_errno(r, "Failed to reset event timer: %m");
+
+        r = sd_event_source_set_time_relative(s, BPF_TIMER_INTERVAL_USEC);
+        if (r < 0)
+                return log_error_errno(r, "Failed to set relative time for timer: %m");
+
+        return 0;
+}
+
+static int monitor_bpf(Manager *m) {
+        _cleanup_(sd_event_source_unrefp) sd_event_source *s = NULL;
+        int r;
+
+        assert(m);
+        assert(m->event);
+
+        r = sd_event_add_time(m->event, &s, CLOCK_MONOTONIC, 0, 0, monitor_bpf_handler, m);
+        if (r < 0)
+                return r;
+
+        r = sd_event_source_set_exit_on_failure(s, true);
+        if (r < 0)
+                return r;
+
+        r = sd_event_source_set_enabled(s, SD_EVENT_ON);
+        if (r < 0)
+                return r;
+
+        (void) sd_event_source_set_description(s, "bpfd-timer");
+
+        m->bpf_timer_event_source = TAKE_PTR(s);
+        return 0;
+}
+
+int manager_new(Manager **ret) {
+        _cleanup_(manager_freep) Manager *m = NULL;
+        int r;
+
+        m = new(Manager, 1);
+        if (!m)
+                return -ENOMEM;
+
+        *m = (Manager) {
+                .enable_logging = true,
+        };
+
+        r = manager_parse_config_file(m);
+        if (r < 0)
+                return r;
+
+        r = sd_event_new(&m->event);
+        if (r < 0)
+                return r;
+
+        r = sd_event_set_signal_exit(m->event, true);
+        if (r < 0)
+                return r;
+
+        r = sd_event_set_watchdog(m->event, true);
+        if (r < 0)
+                log_error_errno(r, "Failed to enable watchdog handling, ignoring: %m");
+
+        *ret = TAKE_PTR(m);
+        return 0;
+}
+
+Manager* manager_free(Manager *m) {
+        int r;
+
+        if (!m)
+                return NULL;
+
+        sd_event_unref(m->event);
+        varlink_server_unref(m->varlink_server);
+
+        r = write_string_file("/proc/sys/kernel/bpf_stats_enabled", "0", WRITE_STRING_FILE_DISABLE_BUFFER);
+        if (r < 0)
+                log_warning_errno(r, "Failed to write '%s' to /proc/sys/kernel/bpf_stats_enabled, ignoring: %m", "0");
+
+
+        return mfree(m);
+}
+
+int manager_start(Manager *m) {
+        int r;
+
+        r = dlopen_bpf();
+        if (r < 0)
+                return log_error_errno(r, "Couldn't dlopen_bpf: %m");
+
+        r = write_string_file("/proc/sys/kernel/bpf_stats_enabled", "1", WRITE_STRING_FILE_DISABLE_BUFFER);
+        if (r < 0)
+                log_warning_errno(r, "Failed to write '%s' to /proc/sys/kernel/bpf_stats_enabled, ignoring: %m", "1");
```

Comment:
you you move this into some helper that just takes a bool and is called one with true and once with false?